azkaban-web-server/src/main/java/azkaban/scheduler/CustomDeserializerDelegate.java

```diff
@@ -0,0 +1,627 @@
+  }
+
+  @Override
+  public int insertFiredTrigger(Connection conn, OperableTrigger trigger, String state, JobDetail job) throws SQLException {
+    return inner.insertFiredTrigger(conn, trigger, state, job);
+  }
+
+  @Override
+  public int updateFiredTrigger(Connection conn, OperableTrigger trigger, String state, JobDetail job) throws SQLException {
+    return inner.updateFiredTrigger(conn, trigger, state, job);
+  }
+
+  @Override
+  public List<FiredTriggerRecord> selectFiredTriggerRecords(Connection conn, String triggerName, String groupName) throws SQLException {
+    return inner.selectFiredTriggerRecords(conn, triggerName, groupName);
+  }
+
+  @Override
+  public List<FiredTriggerRecord> selectFiredTriggerRecordsByJob(Connection conn, String jobName, String groupName) throws SQLException {
+    return inner.selectFiredTriggerRecordsByJob(conn, jobName, groupName);
+  }
+
+  @Override
+  public List<FiredTriggerRecord> selectInstancesFiredTriggerRecords(Connection conn, String instanceName) throws SQLException {
+    return inner.selectInstancesFiredTriggerRecords(conn, instanceName);
+  }
+
+  @Override
+  public Set<String> selectFiredTriggerInstanceNames(Connection conn) throws SQLException {
+    return inner.selectFiredTriggerInstanceNames(conn);
+  }
+
+  @Override
+  public int deleteFiredTrigger(Connection conn, String entryId) throws SQLException {
+    return inner.deleteFiredTrigger(conn, entryId);
+  }
+
+  @Override
+  public int selectJobExecutionCount(Connection conn, JobKey jobKey) throws SQLException {
+    return inner.selectJobExecutionCount(conn, jobKey);
+  }
+
+  @Override
+  public int insertSchedulerState(Connection conn, String theInstanceId, long checkInTime, long interval) throws SQLException {
+    return inner.insertSchedulerState(conn, theInstanceId, checkInTime, interval);
+  }
+
+  @Override
+  public int deleteSchedulerState(Connection conn, String theInstanceId) throws SQLException {
+    return inner.deleteSchedulerState(conn, theInstanceId);
+  }
+
+  @Override
+  public int updateSchedulerState(Connection conn, String theInstanceId, long checkInTime) throws SQLException {
+    return inner.updateSchedulerState(conn, theInstanceId, checkInTime);
+  }
+
+  @Override
+  public List<SchedulerStateRecord> selectSchedulerStateRecords(Connection conn, String theInstanceId) throws SQLException {
+    return inner.selectSchedulerStateRecords(conn, theInstanceId);
+  }
+
+  @Override
+  public Set<String> selectPausedTriggerGroups(Connection conn) throws SQLException {
+    return inner.selectPausedTriggerGroups(conn);
+  }
+
+  @Override
+  protected Object getObjectFromBlob(ResultSet rs, String colName) throws ClassNotFoundException, IOException, SQLException {
+    byte[] bytes = filterBlob(rs.getBytes(colName));
+    if (bytes != null && bytes.length != 0) {
+      InputStream binaryInput = new ByteArrayInputStream(bytes);
+      ObjectInputStream in = new ObjectInputStream(binaryInput);
```

Comment:
If we get to this point, and if we can control which `ObjectInputStream` implementation we use here, then could we possibly implement our own subclass of `ObjectInputStream` that overrides `readClassDescriptor` that effectively ignores `serialVersionUID`, along the lines of https://stackoverflow.com/a/796589? If we do it for one class name and one UID only, then it feels pretty safe. And we don't need to replace bytes, and we don't need to expose the wire format of serialized objects. Any fundamental reasons this won't work?

Reply:
Thanks for the pointer. I've implemented this change.